sql/src/main/java/org/apache/druid/sql/calcite/planner/CalciteRulesManager.java

```diff
@@ -330,6 +331,7 @@ private Program buildReductionProgram(final PlannerContext plannerContext, final
       // make it impossible to convert to COALESCE.
       builder.addRuleInstance(new CaseToCoalesceRule());
       builder.addRuleInstance(new CoalesceLookupRule());
+      builder.addRuleInstance(new RewriteFirstValueLastValueRule());
```

Comment:
It will be good to callout in the description as to what was the need for this new rule. Maybe a simple example with LAST_VALUE and the reason that was not working and how this would rewrite the same query to using a LAG

Reply:
added some description as the apidoc of `RewriteFirstValueLastValueRule` - let me know what you think!